WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverViewController.swift

```diff
@@ -0,0 +1,83 @@
+import UIKit
+import SwiftUI
+import WordPressUI
+
+final class CompliancePopoverViewController: UIViewController {
+
+    // MARK: - Views
+    private lazy var hostingController: UIHostingController = {
+        let controller = UIHostingController(rootView: CompliancePopover(viewModel: self.viewModel))
+        controller.view.translatesAutoresizingMaskIntoConstraints = true
+        return controller
+    }()
+
+    private var contentView: UIView {
+        return hostingController.view
+    }
+
+    private let viewModel: CompliancePopoverViewModel
+    private var bannerIntrinsicHeight: CGFloat = 0
+
+    init(viewModel: CompliancePopoverViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.addContentView()
+        hostingController.rootView.goToSettingsAction = {
+            self.viewModel.didTapSettings()
+        }
+        hostingController.rootView.saveAction = {
+            self.viewModel.didTapSave()
+        }
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        let targetSize = CGSize(width: view.bounds.width, height: 0)
+        let contentViewSize = contentView.systemLayoutSizeFitting(targetSize)
+        self.contentView.frame = .init(origin: .zero, size: contentViewSize)
+        self.preferredContentSize = contentView.bounds.size
+
+        DispatchQueue.main.async {
+            self.hostingController.rootView.screenHeight = self.view.frame.height
+            self.hostingController.rootView.shouldScroll = (contentViewSize.height + 100) > self.view.frame.height
+        }
```

Comment:
Why do we need to push this work to the next run loop? 

Handling layout code this way makes it non-deterministic, which is likely to cause future bugs.

Reply:
In Woo's implementation it mentions that It fixes the issue where `viewDidLayoutSubviews` is called recursively indefinitely. I had the same issue (it was called like 75 thousand times) during the implementation so I kept it. Now I tested it again and it doesn't seem to be happening. It might have been caused by something else in Woo's implementation and after I changed things it went away. I'm not sure. So I'll remove it.